From 695a492c618ada1234bc4484dd3ea66ca9b9b1a4 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Tue, 30 Sep 2014 15:24:13 -0400
Subject: [crypto] qat: Enable interrupts from all 32 bundles

Message-id: <1412090653-14036-1-git-send-email-nikolay@redhat.com>
Patchwork-id: 96804
O-Subject: crypto: qat: Enable interrupts from all 32 bundles
Bugzilla: 1091130
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Bugzilla - https://bugzilla.redhat.com/show_bug.cgi?id=1091130

Description:
 This is an additional small QAT patch that was requested by Intel. It can
 be pulled via the same BZ (thanks Jarod!).
 From: Tadeusz Struk <tadeusz.struk@intel.com>
 Date: Wed, 10 Sep 2014 14:07:25 -0700
 Subject: [RHEL7.1 PATCH BZ 1091130] crypto: qat - Enable all 32 IRQs

 Enable interrupts from all 32 bundles.

 Signed-off-by: Conor McLoughlin <conor.mcloughlin@intel.com>
 Signed-off-by: Tadeusz Struk <tadeusz.struk@intel.com>
 Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
 (cherry picked from commit 413e55742a4f18da6173271ca66748e690997f30)

Upstream:
 413e55742a4f18da6173271ca66748e690997f30

Brew - https://brewweb.devel.redhat.com/taskinfo?taskID=8045144

Tests:
 Compile test.

Signed-off-by: Nikolay Aleksandrov <nikolay@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/crypto/qat/qat_dh895xcc/adf_dh895xcc_hw_data.h b/drivers/crypto/qat/qat_dh895xcc/adf_dh895xcc_hw_data.h
index b707f29..65dd1ff 100644
--- a/drivers/crypto/qat/qat_dh895xcc/adf_dh895xcc_hw_data.h
+++ b/drivers/crypto/qat/qat_dh895xcc/adf_dh895xcc_hw_data.h
@@ -66,7 +66,7 @@
 #define ADF_DH895XCC_ETR_MAX_BANKS 32
 #define ADF_DH895XCC_SMIAPF0_MASK_OFFSET (0x3A000 + 0x28)
 #define ADF_DH895XCC_SMIAPF1_MASK_OFFSET (0x3A000 + 0x30)
-#define ADF_DH895XCC_SMIA0_MASK 0xFFFF
+#define ADF_DH895XCC_SMIA0_MASK 0xFFFFFFFF
 #define ADF_DH895XCC_SMIA1_MASK 0x1
 /* Error detection and correction */
 #define ADF_DH895XCC_AE_CTX_ENABLES(i) (i * 0x1000 + 0x20818)
-- 
1.7.1